### PR TITLE
restructure for modularization. build with jdk11, target 8

### PR DESCRIPTION
**Pull request by pholser**

### CODE REVIEW

File: net.sf.joptsimple/pom.xml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.moditect</groupId>
+                <artifactId>moditect-maven-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (pholser) [reply net.sf.joptsimple/pom.xml:47]: OK, great! I'll apply this learning to some of my other side projects also.

**Comment by pholser** [comment]
> I think you could improve the build setup. I'm a little bit confused why you split the build into two maven modules, for the core and tests. My guess is that would only make testing in a modular setting more difficult. Did you run into issues trying to access test libraries?

Nah, just figured I'd work toward having the tests know only about the exported stuff from the main module. Not quite there yet ... I'll implement your other suggestions first.

Thanks for the feedback! I'm learning a lot by undertaking this modularization.

**Comment by pholser** [comment]
@A248 I appreciate the time you took to offer feedback on this PR.

Quick question: it's great that the maven-compiler-plugin can be made to compile a module-info.java file though we may be targeting JDK 8 for the bytecode. Should I be making a multi-release JAR, with the module-info tucked under `META-INF/versions/9`? Or is the blurb in the moditect docs about class file scanners choking on a module-info.class in a JAR such as mine overblown?

**Comment by pholser** [comment]
@A248 Thanks again for your feedback. I've updated the configs and some tests in the process.

**Comment by pholser** [comment]
> Also, sorry if I seem a little snappy, that's just how I sound some times. You sound a lot friendlier by comparison.

@A248 No worries, I didn't get that sense at all. Appreciate the learning opportunities!

At your convenience, have a look at recent updates to this PR. I think we're converging on a satisfactory solution.

**Review by A248 — APPROVED** [review]
It all looks good to me. I'm sure you are aware the module name you chose is different from the previous Automatic-Module-Name, but that's fine since 6.0 is a new major release.

Also, you are dropping OSGI support, which I personally know little to nothing about.

**Comment by pholser** [comment]
@A248 Thanks! re: OSGi -- you and me both. I suppose OSGi and JPMS metadata can coexist in a JAR, not sure how to get there just yet.

**Pull request merged**